twig/compiler.py

```diff
@@ -1,21 +1,25 @@
+from pathlib import Path
+from typing import Any, Dict, Iterable
+
+from twig.backends import VyperBackend
 from twig.utils.compiler import generate_contract_types, generate_inline_sources
 
 
 class Compiler:
-    def __init__(self, sources, backend):
+    def __init__(self, sources: Iterable[Path], backend: VyperBackend) -> None:
```

Comment:
Same here on input type.  `Sequence` is our goto unless we explicitely support generators which we do occasionally, but usually only for internal APIs since they can result in un-intended errors if the code tries to consume the generator more than once.